src/controller_plugin.py

```diff
@@ -199,20 +199,153 @@ def get_job_id(job):
         """
 
         return job["id"]
+
+    @staticmethod
+    def get_status(job):
+        """returns a job's status
+
+        Arguments:
+            job {dict} -- A dict in the format of a job
+
+        Returns:
+            str -- the job's current status
+        """
+
+        return job["Status"]
     
     @staticmethod
-    def value_of_input(job, input):
-        try:
-            return job["JobCommand"]["Inputs"][input]["Value"]
-        except IndexError:
+    def value_of(job, input):
+        """returns the value of an input name
+
+        Arguments:
+            job {dict} -- A dict in the format of a job
+            input {str} -- the name of an input or optional input
+
+        Returns:
+            str -- The value of the first input or optional input with the
+            given name. If there is an input or an optional input with the
+            same name, the input's value will be returned. None if no inputs
+            found.
+        """
+
+        if isinstance(input, str):
+            value = ControllerPlugin.value_of_input(job, input)
+            if value is None:
+                value = ControllerPlugin.value_of_option(job, input)
+            return value
+        else:
             return None
 
+    @staticmethod
+    def value_of_input(job, option):
+        """Get the value of an input by index or name
+
+        Arguments:
+            job {dict} -- A dict in the format of a job
+            option {int|str} -- The index of an input or the name of an input.
+
+        Returns:
+            str|None -- The value of the given input. None if no input found.
+        """
+
+        return ControllerPlugin._srch_4_val(
+            job["JobCommand"]["Inputs"],
+            option
+        )
+
     @staticmethod
     def value_of_option(job, option):
+        """Get the value of an optional input by index or name
+
+        Arguments:
+            job {dict} -- A dict in the format of a job
+            input {int|str} -- The index of an input or the name of an input.
+
+        Returns:
+            str|None -- The value of the given input. None if no input found.
+        """
+
+        return ControllerPlugin._srch_4_val(
+            job["JobCommand"]["OptionalInputs"],
+            option
+        )
+
+    @staticmethod
+    def _srch_4_val(val_list, search):
         try:
-            return job["JobCommand"]["OptionalInputs"][option]["Value"]
+            return val_list[search]["Value"]
         except IndexError:
             return None
+        except TypeError:
+            for i in val_list:
+                if i["Name"] == search:
+                    return i["Value"]
+        return None
+
+    @staticmethod
+    def get_args(job):
```

Comment:
Are these necessary? Seems like a lot extra when they could just access the lists in the dictionary.